### PR TITLE
ENG-1867 Use Vite in common

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR migrates common from tsdx to use vite as it's build tool

## Related issue number (if any)
ENG-1867

## Loom demo (if any)
https://www.loom.com/share/2d7e227921d14a5fac7ecc684b69c435

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/package.json [hunk 1]
```diff
@@ -17,9 +17,12 @@
   "scripts": {
     "preinstall": "npx force-resolutions",
     "start": "tsdx watch",
-    "build": "tsdx build",
+    "build": "vite build",
+    "dev": "vite",
```
Comment (agiron123) [inline src/ui/common/package.json:21]: change to start

File: src/ui/common/package.json [hunk 2]
```diff
@@ -17,9 +17,12 @@
...
+    "buildvite": "tsc && vite build",
+    "preview": "vite preview",
     "test": "tsdx test --passWithNoTests",
-    "prepare": "tsdx build",
+    "prepare": "npm run build",
```
Comment (agiron123) [inline src/ui/common/package.json:25]: vite build

File: src/ui/common/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import react from '@vitejs/plugin-react';
+
+// https://vitejs.dev/config/
+// export default defineConfig(({ command, mode }) => {
+//     plugins: [react()],
+
+// })
+
+export default defineConfig(({ command, mode }) => {
+    if (command === 'serve') {
```
Comment (agiron123) [inline src/ui/common/vite.config.ts:13]: don't need these if statements since we're just logging here.

**Comment by vsreekanti** [comment]
Neat! A couple small things:
* Does `vite` have an equivalent of `tsdx watch` that we can use for development? 
* Can we remove the old tsdx config files?

**Review by vsreekanti — APPROVED** [review]

**Comment by agiron123** [comment]
> Neat! A couple small things:
> 
> * Does `vite` have an equivalent of `tsdx watch` that we can use for development?
> * Can we remove the old tsdx config files?

@vsreekanti yes, it's just the vite command.

**Comment by agiron123** [comment]
Created followup issue to address using vite in app:
https://linear.app/aqueducthq/issue/ENG-1904/use-vite-in-app-build

**Pull request merged**